services/autoscaling/groups.go

```diff
@@ -0,0 +1,544 @@
+	}
+
+	return &groupPolicy
+}
+
+func GetScaleInActionData(d *schema.ResourceData) *autoscaling.GroupPolicyScaleInAction {
+	scaleInAction := autoscaling.GroupPolicyScaleInAction{}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.amount"); ok {
+		value := float32(value.(int))
+		scaleInAction.Amount = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.amount_type"); ok {
+		value := autoscaling.ActionAmount(value.(string))
+		scaleInAction.AmountType = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.termination_policy_type"); ok {
+		value := autoscaling.TerminationPolicyType(value.(string))
+		scaleInAction.TerminationPolicy = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.cooldown_period"); ok {
+		value := value.(string)
+		scaleInAction.CooldownPeriod = &value
+	} else {
+		scaleInAction.CooldownPeriod = nil
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.delete_volumes"); ok {
+		value := value.(bool)
+		scaleInAction.DeleteVolumes = &value
+	} else {
+		scaleInAction.DeleteVolumes = nil
+	}
+
+	return &scaleInAction
+}
+
+func GetScaleOutActionData(d *schema.ResourceData) *autoscaling.GroupPolicyScaleOutAction {
+	scaleOutAction := autoscaling.GroupPolicyScaleOutAction{}
+
+	if value, ok := d.GetOk("policy.0.scale_out_action.0.amount"); ok {
+		value := float32(value.(int))
+		scaleOutAction.Amount = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_out_action.0.amount_type"); ok {
+		value := autoscaling.ActionAmount(value.(string))
+		scaleOutAction.AmountType = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_out_action.0.cooldown_period"); ok {
+		value := value.(string)
+		scaleOutAction.CooldownPeriod = &value
+	} else {
+		scaleOutAction.CooldownPeriod = nil
+	}
+
+	return &scaleOutAction
+}
+
+func GetReplicaConfigurationPostData(d *schema.ResourceData) (*autoscaling.ReplicaPropertiesPost, error) {
+	replica := autoscaling.ReplicaPropertiesPost{}
+
+	if value, ok := d.GetOk("replica_configuration.0.availability_zone"); ok {
+		value := autoscaling.AvailabilityZone(value.(string))
+		replica.AvailabilityZone = &value
+	}
+
+	if value, ok := d.GetOk("replica_configuration.0.cores"); ok {
+		value := int32(value.(int))
+		replica.Cores = &value
+	}
+
+	if value, ok := d.GetOk("replica_configuration.0.cpu_family"); ok {
+		value := autoscaling.CpuFamily(value.(string))
+		replica.CpuFamily = &value
+	} else {
+		replica.CpuFamily = nil
+	}
+
+	replica.Nics = GetNicsData(d)
+	if *replica.Nics == nil {
+		*replica.Nics = make([]autoscaling.ReplicaNic, 0)
+	}
+
+	if value, ok := d.GetOk("replica_configuration.0.ram"); ok {
+		value := int32(value.(int))
+		replica.Ram = &value
+	}
+
+	if volumes, err := GetVolumesData(d); err != nil {
+		return nil, err
+	} else {
+		replica.Volumes = volumes
+	}
+
+	return &replica, nil
+}
+
+func GetNicsData(d *schema.ResourceData) *[]autoscaling.ReplicaNic {
+	var nics []autoscaling.ReplicaNic
```

Comment:
Can use `nics := make([]autoscaling.ReplicaNic,0` instead of `var` declaration in **GetNicsData**
It will eliminate the need to perform the nilcheck afterwards

Reply:
@cbeti-ionos solved in c9df22e0066ca69a7809de59cd124fe97fec9580